src/microbe_stage/editor/EditorActionHistory.cs

```diff
@@ -138,8 +138,16 @@ public override bool Undo()
 
     public override void AddAction(TAction action)
     {
-        // TODO: check if the action can be combined (for example behaviour or rigidity slider subsequent edits should
+        // Check if the action can be combined (for example behaviour or rigidity slider subsequent edits should
         // combine) in a single step for undo
+        if (ActionIndex > 0 && action.Data.FirstOrDefault()!.WantsMerge)
+        {
+            EditorCombinableActionData currentActionData = action.Data.FirstOrDefault()!;
+            EditorCombinableActionData lastActionData = Actions[ActionIndex - 1].Data.FirstOrDefault()!;
+
+            if (currentActionData.TryMerge(lastActionData))
+                PopTopAction();
```

Comment:
Wouldn't it be easier to reason about this if this was flipped. So that the already performed action is the one that gets merged with the subsequent action? This is because that's the order in which MP computation does it, so if this does it the same way around, it'll be less error prone for future programmers to work on this.